tensorflow/python/keras/layers/convolutional_test.py

```diff
@@ -54,11 +54,15 @@ def _run_test(self, kwargs, expected_output_shape):
       ('padding_causal', {'padding': 'causal'}, (None, 7, 2)),
       ('strides', {'strides': 2}, (None, 3, 2)),
       ('dilation_rate', {'dilation_rate': 2}, (None, 3, 2)),
+      # Only runs on GPU with CUDA, groups are not supported on CPU.
+      # https://github.com/tensorflow/tensorflow/issues/29005
+      ('group', {'groups': 3, 'filters': 6}, (None, 5, 6), True),
```

Comment:
- Please add tests for the error messages added
- Please add a correctness test; current tests only check the output shape, not whether group convolution has actually taken place.

Reply:
> Please add tests for the error messages added

:+1: I added tests for this in the latest commits

> Please add a correctness test; current tests only check the output shape, not whether group convolution has actually taken place.

I looked at the tests for Keras convolutional layers and the `tf.nn` and none of them test the actual correctness of the output. The testing for correctness of the grouped convolution and normal convolutions is already handled on the C++.
I can add a very simple test that checks that this implementation matches a simple loop based version of grouped convolutions. Do you think that would be useful?

Reply:
> Please add a correctness test; current tests only check the output shape, not whether group convolution has actually taken place.

I added more tests in 2c703ebe0b3dd7ec0da4017be54cb15d268c5638